controllers/datadogagent/feature/dogstatsd/feature.go

```diff
@@ -0,0 +1,198 @@
+
+	udsEnabled      bool
+	udsHostFilepath string
+
+	originDetectionEnabled bool
+	mapperProfiles         *apicommonv1.CustomConfig
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *dogStatsDFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	dogstatsd := dda.Spec.Features.Dogstatsd
+	if apiutils.BoolValue(dogstatsd.HostPortConfig.Enabled) {
+		f.hostPortEnabled = true
+	}
+	if dogstatsd.HostPortConfig.Port != nil {
+		f.hostPortHostPort = *dogstatsd.HostPortConfig.Port
+	}
+	if apiutils.BoolValue(dogstatsd.UnixDomainSocketConfig.Enabled) {
+		f.udsEnabled = true
+	}
+	f.udsHostFilepath = *dogstatsd.UnixDomainSocketConfig.Path
+	if apiutils.BoolValue(dogstatsd.OriginDetectionEnabled) {
+		f.originDetectionEnabled = true
+	}
+	if dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v2alpha1.ConvertCustomConfig(dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
+		},
+	}
+	return reqComp
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *dogStatsDFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
+	config := dda.Spec.Agent.Config
+	f.hostPortEnabled = true
+	if config.HostPort != nil {
+		f.hostPortHostPort = *config.HostPort
+	}
+	if apiutils.BoolValue(config.Dogstatsd.UnixDomainSocket.Enabled) {
+		f.udsEnabled = true
+	}
+	if config.Dogstatsd.UnixDomainSocket.HostFilepath != nil {
+		f.udsHostFilepath = *config.Dogstatsd.UnixDomainSocket.HostFilepath
+	}
+	if apiutils.BoolValue(config.Dogstatsd.DogstatsdOriginDetection) {
+		f.originDetectionEnabled = true
+	}
+	if config.Dogstatsd.MapperProfiles != nil {
+		f.mapperProfiles = v1alpha1.ConvertCustomConfig(config.Dogstatsd.MapperProfiles)
+	}
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
+			},
+		},
+	}
+	return reqComp
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *dogStatsDFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *dogStatsDFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// udp
+	if f.hostPortEnabled {
+		if f.hostPortHostPort != 0 {
```

Comment:
if `f.hostPortEnabled == true` then `f.hostPortHostPort` should always have a nonzero value, so i think the `else` block can be removed (no need to do defaulting here)